activerecord/lib/active_record/associations.rb

```diff
@@ -229,7 +229,16 @@ def association(name) #:nodoc:
       association = association_instance_get(name)
 
       if association.nil?
-        unless reflection = self.class._reflect_on_association(name)
+        reflection = self.class._reflect_on_association(name)
```

Comment:
Should I push this down the stack to `_reflect_on_association`? The problem there, is that since this method is included via an `AS::Concern`, it doesn't know about the current class its dealing with, and thus wouldn't be able to construct the superclass tree, etc.